bd-logger/src/async_log_buffer.rs

```diff
@@ -866,6 +879,21 @@ impl<R: LogReplay + Send + 'static> AsyncLogBuffer<R> {
               log::debug!("failed to process crash log: {e}");
             }
           }
+
+          if crash_monitor.is_reports_watcher_enabled() {
+            self.start_file_watcher_if_needed(&crash_monitor);
+          }
+        },
+        file_path_opt = async {
+          if let Some(rx) = self.current_session_file_rx.as_mut() {
+            rx.recv().await
+          } else {
+            future::pending().await
+          }
```

Comment:
Maybe I'm missing something but why do we need to go back and forth between the async log buffer and the crash handler to support this? If we're having to do this to poll the futures we should probably just do a new task in the crash handler